common/lib/xmodule/xmodule/js/src/video/08_video_speed_control.js

```diff
@@ -24,24 +28,51 @@ function (Iterator) {
     };
 
     SpeedControl.prototype = {
+        template: [
+            '<div class="speeds menu-container">',
+                '<a class="speed-button" href="#" title="',
+                    gettext('Speeds'), '" role="button" aria-disabled="false">',
+                    '<span class="label">', gettext('Speed'), '</span>',
+                    '<span class="value"></span>',
+                '</a>',
+              '<ol class="video-speeds menu" role="menu"></ol>',
+            '</div>'
+        ].join(''),
+
+        destroy: function () {
+            this.el.off({
+                'mouseenter': this.mouseEnterHandler,
+                'mouseleave': this.mouseLeaveHandler,
+                'click': this.clickMenuHandler,
+                'keydown': this.keyDownMenuHandler
+            });
+
+            this.state.el.off({
+                'speed:set': this.onSetSpeed,
+                'speed:render': this.onRenderSpeed
+            });
+            this.closeMenu(true);
+            this.speedsContainer.remove();
+            this.el.remove();
+            delete this.state.videoSpeedControl;
+        },
+
         /** Initializes the module. */
         initialize: function () {
             var state = this.state;
 
-            this.el = state.el.find('.speeds');
-            this.speedsContainer = this.el.find('.video-speeds');
-            this.speedButton = this.el.find('.speed-button');
-
             if (!this.isPlaybackRatesSupported(state)) {
-                this.el.remove();
                 console.log(
                     '[Video info]: playbackRate is not supported.'
                 );
 
                 return false;
             }
-
+            this.el = $(this.template);
+            this.speedsContainer = this.el.find('.video-speeds');
+            this.speedButton = this.el.find('.speed-button');
             this.render(state.speeds, state.speed);
+            this.setSpeed(state.speed, true, true);
```

Comment:
> Doesn't setSpeed take only one parameter now?

it's local for the module setSpeed method (it is not the same as this.state.setSpeed).